stdlib/array.gr

```diff
@@ -892,3 +895,49 @@ export let sort = (comp, array) => {
   let len = length(array)
   quicksort(0, len - 1)
 }
+
+/**
+ * Rotates an array by n elements to the right, in place.
+ *
+ * If n is negative, the array will be rotated by n elements
+ * to the left. See examples.
+ *
+ * @param arr: The array to be rotated
+ * @param n: The number of elements to rotate by
+ * @returns The rotated array
+ *
+ * @example: rotate([> 1, 2, 3, 4, 5], 2) == [> 4, 5, 1, 2, 3]
+ * @example: rotate([> 1, 2, 3, 4, 5], -1) == [> 2, 3, 4, 5, 1]
```

Comment:
```suggestion
 * @example let array = [> 1, 2, 3, 4, 5]; rotate(2, arr); arr == [> 4, 5, 1, 2, 3]
 * @example let array = [> 1, 2, 3, 4, 5]; rotate(-1, arr); arr == [> 2, 3, 4, 5, 1]
```
A little messy on these examples, but that's the only way to do it if we don't return an array.